functional-samples/cookbook.offscreen-dom/background.js

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Close any existing offscreen documents to avoid inconsistencies
+// with an updated service worker version
+addEventListener('install', async (event) => {
+  closeOffscreenDocument();
```

Comment:
@oliverdunk not sure if this is really needed, but there were cases (I think) in which sw and offscreen doc were out of sync.

Reply:
I haven't seen/done this before, but given offscreen documents are new it's possible there's an edge none of us have noticed. I'd be curious if you have steps to reproduce or a video of what you're seeing when this isn't present.

I haven't personally done any testing around extension updates so it's possible the offscreen documents doesn't get removed I guess.